framework/include/utils/RankFourTensorImplementation.h

```diff
@@ -0,0 +1,894 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "RankFourTensor.h"
+
+// MOOSE includes
+#include "RankTwoTensor.h"
+#include "MooseEnum.h"
+#include "MooseException.h"
+#include "MooseUtils.h"
+#include "MatrixTools.h"
+#include "PermutationTensor.h"
+
+#include "libmesh/utility.h"
+#include "libmesh/tensor_value.h"
+#include "libmesh/vector_value.h"
+
+// C++ includes
+#include <iomanip>
+#include <ostream>
+
+namespace MathUtils
+{
+template <>
+void mooseSetToZero<RankFourTensorTempl<Real>>(RankFourTensorTempl<Real> & v);
+template <>
+void mooseSetToZero<RankFourTensorTempl<DualReal>>(RankFourTensorTempl<DualReal> & v);
```

Comment:
These are repeat declarations that are already in `RankFourTensor.h`. So you can remove them if you would like. It's not doing any harm here though, so I won't require it :smile: The same is true for the analogs in `RankThreeTensor` and `RankFourTensor`